### PR TITLE
Small changes to support releases

### PR DESCRIPTION
**Pull request by joerunde**
These are some changes we've been adding onto the last couple releases. Would be nice to have on main for the next round, and hopefully this is all moot once we hop over to the ODH fork anyway

### CODE REVIEW

File: vllm/entrypoints/grpc/grpc_server.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def __init__(self, engine: AsyncLLMEngine, args: argparse.Namespace):
 
     async def _post_init(self):
         self.config = await self.engine.get_model_config()
-        self.tokenizer_group = await self.engine.get_tokenizer_group()
+         # self.tokenizer_group = await self.engine.get_tokenizer_group()
+        self.tokenizer_group = self.engine.engine.tokenizer
```
Comment (njhill) [inline vllm/entrypoints/grpc/grpc_server.py:122]: I actually don't think this is required anymore with the latest upstream changes.

**Comment by njhill** [comment]
They also just removed the custom vllm-nccl so we should remove the relevant part of our Dockerfile for that too (also fyi @tjohnson31415)

File: Dockerfile.ubi [hunk 2]
```diff
@@ -193,7 +193,7 @@ ENV TORCH_CUDA_ARCH_LIST=$TORCH_CUDA_ARCH_LIST
 ARG max_jobs=2
 ENV MAX_JOBS=${max_jobs}
 # number of threads used by nvcc
-ARG nvcc_threads=8
+ARG nvcc_threads=2
```
Comment (dtrifiro) [inline Dockerfile.ubi:196]: Depending on the hardwware you have, setting nvcc_threads=2 and max_jobs=2 will result in 1 job running two nvcc_threads (max jobs is overridden to `max(1, max_jobs//nvcc_threads)`. Note that each nvcc thread can use up to ~3.5GB of memory.

File: Dockerfile.ubi [hunk 3]
```diff
@@ -217,7 +222,7 @@ COPY --from=gen-protos --link /workspace/vllm/entrypoints/grpc/pb vllm/entrypoin
 ENV CCACHE_DIR=/root/.cache/ccache
 RUN --mount=type=cache,target=/root/.cache/ccache \
     --mount=type=cache,target=/root/.cache/pip \
-    VLLM_USE_PRECOMPILED=1 python3 setup.py bdist_wheel --dist-dir=dist
+    python3 setup.py bdist_wheel --dist-dir=dist
```
Comment (dtrifiro) [inline Dockerfile.ubi:225]: You're not stripping/optimizing the built binaries here, see this https://github.com/opendatahub-io/vllm/pull/30/files for the correct cmake argument

File: vllm/entrypoints/grpc/grpc_server.py (hunk 1 above)
Comment (dtrifiro) [reply vllm/entrypoints/grpc/grpc_server.py:122]: Indeed: https://github.com/vllm-project/vllm/blob/main/vllm/engine/llm_engine.py#L381-L387

File: Dockerfile.ubi (hunk 3 above)
Comment (joerunde) [reply Dockerfile.ubi]: Thanks!

File: Dockerfile.ubi (hunk 2 above)
Comment (joerunde) [reply Dockerfile.ubi:196]: yeah it seems like anything above this just craps out in GHA anyway :(

Slow builds gonna be slow 🐢

**Comment by joerunde** [comment]
Just sliding this in here for the next lil' release

**Pull request merged**